### PR TITLE
[Drata] Part 3 - Remaining Triggers

### PR DESCRIPTION
**Pull request by andrewjschuang**
Part 3 out 3 of #5139.

Triggers:

- [X] Failed Monitor
- [ ] New Framework
- [ ] Audit Hub

### <samp>🤖 Generated by Copilot at 4a771a5</samp>

This pull request adds new action and source components for the Drata app, as well as common files for constants, utilities, and the app definition. It also updates the `package.json` file and deletes unused files. The new components allow users to create, update, find, and upload data for various Drata entities, such as assets, controls, vendors, personnel, and documents. The new sources emit events for changes in the Drata audit hub, control status, and asset and control creation.

### <samp>🤖 Generated by Copilot at 4a771a5</samp>

> _Drata components_
> _Actions, sources, and methods_
> _Autumn of `drata.app`_

### <samp>🤖 Generated by Copilot at 4a771a5</samp>

*  Add new action components for Drata ([link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-c86e1662c8e6d8ed39e41867f186a58ee916f671d695a798086c46bb3c1db5bcR1-R105), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-5d59c9a53ce9ac90d972c2a7ffeeea7ef1e5c0f52395ccac5445e6d828a026f4R1-R62), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-6bad81e2ede0f4e8c3e655e9a886b941a02d5b50b2b087a6eabc4595a1170eb0R1-R219), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-ef022ab6f2cd79eb416d8709ab76cfad33d9bc73b76e5a40367d3e63d84440c0R1-R138), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-6fc3124e161dfdbfcbd3ba77757250d1a89eb68343fd24d7d68f931a1a9776e3R1-R142), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-cc630ee99b563f077851b2d43e2e653600356339ed85694c9fa96db2429fcbe0R1-R225), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-650a63536f174fe90eba52f7955560ec822dac79d90a63a4bf0cd3b984e63c20R1-R56), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-60b002040098c47c885a597dacd638b9ee7ca5bc763e353ae519b547e798da02R1-R111), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-eddcddc1253db63df45f9fa1042bd6e7f6e3041cf783a713fd1839f5430853ddR1-R111))
* Add new source components for Drata ([link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-cbed5011f69f658467a1a830011ff832f4ffb161b3be5344ae316eae66cedbf1R1-R28), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-4e4fdc5f8037aa7f0630bbb1665042de8dde21f7bea619f5cc10541e09ab336aR1-R28), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-cae736d151093a9eb01cf81fb7fb4248c39c6dbc2dd9c274397792585de1e21bR1-R86), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-6449b3851ac7649bb15eb4ece7ee9c9702cf24eb5f13950a31546dd2f2815eabR1-R75))
* Add a new file for defining the Drata app ([link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-a07058fe354e5714b5b288324099660f299a6c49cb5ec445a7251a3072c43db2R1-R269))
* Add a new file for defining common constants for the Drata components ([link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-ad40b68422e9aa2a165ae2c416fc624458a76e16b6e166ea2bd7cf215e4d56f0R1-R10))
* Add a new file for defining common utility methods for the Drata components ([link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-557f97192f075cec88be2ca98c4d90a2f89af57788470577936f1a6f2b9b4106R1-R22))
* Update the `package.json` file for the Drata components ([link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-f308bcfbd8557663c1c9360bb4d8bc09ad1946ad1dfe5c5bc638abe738126f45L3-R17))
* Delete the unused files for the Drata components ([link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-7506fdc3e6d938478e697eeeb6ce05845536e1a978ede9657834c9549fdb79d0), [link](https://github.com/PipedreamHQ/pipedream/pull/6028/files?diff=unified&w=0#diff-27405f52b709f4496e9f16016505e167d7bc6d11623b4451d89abcea14a553d5))

### CODE REVIEW

File: components/drata/sources/audit-hub/audit-hub.mjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import drata from "../../drata.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+const docsLink = "";
+
+export default {
+  key: "drata-audit-hub",
+  name: "Audit Hub",
+  description: `Emit new event for a audit hub result. [See docs here.](${docsLink})`,
```
Comment (dylburger) [inline components/drata/sources/audit-hub/audit-hub.mjs:9]:
```suggestion
  description: `Emit a new event for an audit hub result. [See docs here.](${docsLink})`,
```

Comment (dylburger) [reply components/drata/sources/audit-hub/audit-hub.mjs:9]: Changed "Emit new event" to "Emit a new event" to follow Google's Developer docs style guide (use articles).

File: components/drata/sources/failed-monitor/failed-monitor.mjs [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import drata from "../../drata.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/MonitorsPublicController_listMonitors/";
+
+export default {
+  key: "drata-failed-monitor",
+  name: "Failed Monitor",
+  description: `Emit new event whenever a monitor fails. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/failed-monitor/failed-monitor.mjs:9]:
```suggestion
  description: `Emit a new event whenever a monitor fails. [See the docs here](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/failed-monitor/failed-monitor.mjs:9]: Changed 'new event' to 'a new event' and 'docs' to 'the docs' to follow Google's Developer docs style guide.

File: components/drata/sources/failed-monitor/failed-monitor.mjs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+    checkType: {
+      type: "string",
+      label: "Check Type",
+      description: "Filter monitors by check type (associated monitor instances)",
```
Comment (dylburger) [inline components/drata/sources/failed-monitor/failed-monitor.mjs:25]:
```suggestion
      description: "Filter monitors by check type (associated monitor instances)",
```

Comment (dylburger) [reply components/drata/sources/failed-monitor/failed-monitor.mjs:25]: Removed unnecessary parentheses to follow Dreyer's English.

File: components/drata/sources/failed-monitor/failed-monitor.mjs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      });
+
+      const visitedIds = {};
+      for (const monitor of response.data) {
+        const ts = Date.parse(monitor.lastCheck);
+        visitedIds[monitor.id] = ts;
+
+        this.$emit(monitor, {
+          id: `${monitor.id}_${ts}`,
+          summary: `Historical failed monitor event: ${monitor.name}`,
```
Comment (dylburger) [inline components/drata/sources/failed-monitor/failed-monitor.mjs:57]:
```suggestion
          summary: `Historical failed monitor event: ${monitor.name}`,
```

Comment (dylburger) [reply components/drata/sources/failed-monitor/failed-monitor.mjs:57]: Changed 'Historical' to 'historical' to follow Google's Developer docs style guide."

File: components/drata/sources/failed-monitor/failed-monitor.mjs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    for (const monitor of response.data) {
+      const id = monitor.id;
+      const ts = Date.parse(monitor.lastCheck);
+
+      if (!visitedIds[id] || ts > visitedIds[id]) {
+        visitedIds[id] = ts;
+
+        this.$emit(monitor, {
+          id: `${monitor.id}_${ts}`,
+          summary: `Failed: ${monitor.name}`,
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/drata/sources/failed-monitor/failed-monitor.mjs:93]: Changed 'Failed' to 'failed' to follow Google's Developer docs style guide."

File: components/drata/sources/failed-monitor/failed-monitor.mjs [hunk 6]
```diff
@@ -0,0 +1,101 @@
+import drata from "../../drata.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/MonitorsPublicController_listMonitors/";
+
+export default {
+  key: "drata-failed-monitor",
+  name: "Failed Monitor",
+  description: `Emit a new event whenever a monitor fails. [See docs here](${docsLink}).`,
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/drata/sources/failed-monitor/failed-monitor.mjs:9]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/new-asset-added/new-asset-added.mjs [hunk 7]
```diff
@@ -7,7 +7,7 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-asset-added",
   name: "New Asset Added",
-  description: `Emit new event for every new asset. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new asset. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-asset-added/new-asset-added.mjs:10]:
```suggestion
  description: `Emit a new event for every new asset. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/new-asset-added/new-asset-added.mjs:10]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/new-control-added/new-control-added.mjs [hunk 8]
```diff
@@ -7,7 +7,7 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-control-added",
   name: "New Control Added",
-  description: `Emit new event for every new control. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new control. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-control-added/new-control-added.mjs:10]:
```suggestion
  description: `Emit a new event for every new control. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/new-control-added/new-control-added.mjs:10]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/new-evidence-added/new-evidence-added.mjs [hunk 9]
```diff
@@ -7,7 +7,7 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-evidence-added",
   name: "New Evidence for Control Added",
-  description: `Emit new event for every new evidence for a control. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new evidence for a control. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-evidence-added/new-evidence-added.mjs:10]:
```suggestion
  description: `Emit a new event for every new evidence for a control. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/new-evidence-added/new-evidence-added.mjs:10]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/new-personnel-added/new-personnel-added.mjs [hunk 10]
```diff
@@ -7,7 +7,7 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-personnel-added",
   name: "New Personnel Added",
-  description: `Emit new event for every new personnel. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new personnel. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-personnel-added/new-personnel-added.mjs:10]:
```suggestion
  description: `Emit a new event for every new personnel. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/new-personnel-added/new-personnel-added.mjs:10]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/new-vendor-added/new-vendor-added.mjs [hunk 11]
```diff
@@ -7,7 +7,7 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-vendor-added",
   name: "New Vendor Added",
-  description: `Emit new event for every new vendor. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new vendor. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-vendor-added/new-vendor-added.mjs:10]:
```suggestion
  description: `Emit a new event for every new vendor. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/new-vendor-added/new-vendor-added.mjs:10]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/terminated-employee/terminated-employee.mjs [hunk 12]
```diff
@@ -7,7 +7,7 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-terminated-employee",
   name: "Employee Terminated",
-  description: `Emit new event when an employee is terminated. [See docs here](${docsLink}).`,
+  description: `Emit a new event when an employee is terminated. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/terminated-employee/terminated-employee.mjs:10]:
```suggestion
  description: `Emit a new event when an employee is terminated. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/terminated-employee/terminated-employee.mjs:10]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/vendor-updated/vendor-updated.mjs [hunk 13]
```diff
@@ -7,7 +7,7 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-vendor-updated",
   name: "Vendor Updated",
-  description: `Emit new event when a vendor is updated. [See docs here](${docsLink}).`,
+  description: `Emit a new event when a vendor is updated. [See docs here](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/vendor-updated/vendor-updated.mjs:10]:
```suggestion
  description: `Emit a new event when a vendor is updated. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/vendor-updated/vendor-updated.mjs:10]: Changed "docs here" to "the documentation" to follow Google's Developer docs style guide.

File: components/drata/sources/failed-monitor/failed-monitor.mjs [hunk 14]
```diff
@@ -0,0 +1,101 @@
+import drata from "../../drata.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/MonitorsPublicController_listMonitors/";
+
+export default {
+  key: "drata-failed-monitor",
+  name: "Failed Monitor",
+  description: `Emit a new event whenever a monitor fails. [See the documentation](${docsLink}).`,
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/drata/sources/failed-monitor/failed-monitor.mjs:9]: Removed "new" for consistency with Google's Developer docs style guide

File: components/drata/sources/new-asset-added/new-asset-added.mjs [hunk 15]
```diff
@@ -7,9 +7,9 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-asset-added",
   name: "New Asset Added",
-  description: `Emit new event for every new asset. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new asset. [See the documentation](${docsLink}).`,
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/drata/sources/new-asset-added/new-asset-added.mjs:10]: Removed "new" for consistency with Google's Developer docs style guide

File: components/drata/sources/new-control-added/new-control-added.mjs [hunk 16]
```diff
@@ -7,9 +7,9 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-control-added",
   name: "New Control Added",
-  description: `Emit new event for every new control. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new control. [See the documentation](${docsLink}).`,
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/drata/sources/new-control-added/new-control-added.mjs:10]: Removed "new" for consistency with Google's Developer docs style guide

File: components/drata/sources/new-evidence-added/new-evidence-added.mjs [hunk 17]
```diff
@@ -7,9 +7,9 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-evidence-added",
   name: "New Evidence for Control Added",
-  description: `Emit new event for every new evidence for a control. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new evidence for a control. [See the documentation](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-evidence-added/new-evidence-added.mjs:10]:
```suggestion
  description: `Emit an event for every new evidence for a control. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/new-evidence-added/new-evidence-added.mjs:10]: Removed "new" for consistency with Google's Developer docs style guide

File: components/drata/sources/new-personnel-added/new-personnel-added.mjs [hunk 18]
```diff
@@ -7,9 +7,9 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-personnel-added",
   name: "New Personnel Added",
-  description: `Emit new event for every new personnel. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new personnel. [See the documentation](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-personnel-added/new-personnel-added.mjs:10]:
```suggestion
  description: `Emit an event for every new personnel. [See the documentation](${docsLink}).`,
```

File: components/drata/sources/new-vendor-added/new-vendor-added.mjs [hunk 19]
```diff
@@ -7,9 +7,9 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-new-vendor-added",
   name: "New Vendor Added",
-  description: `Emit new event for every new vendor. [See docs here](${docsLink}).`,
+  description: `Emit a new event for every new vendor. [See the documentation](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/new-vendor-added/new-vendor-added.mjs:10]:
```suggestion
  description: `Emit an event for every new vendor. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/new-vendor-added/new-vendor-added.mjs:10]: Removed "new" for consistency with Google's Developer docs style guide

File: components/drata/sources/terminated-employee/terminated-employee.mjs [hunk 20]
```diff
@@ -7,9 +7,9 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-terminated-employee",
   name: "Employee Terminated",
-  description: `Emit new event when an employee is terminated. [See docs here](${docsLink}).`,
+  description: `Emit a new event when an employee is terminated. [See the documentation](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/terminated-employee/terminated-employee.mjs:10]:
```suggestion
  description: `Emit an event when an employee is terminated. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/terminated-employee/terminated-employee.mjs:10]: Removed "new" for consistency with Google's Developer docs style guide

File: components/drata/sources/vendor-updated/vendor-updated.mjs [hunk 21]
```diff
@@ -7,9 +7,9 @@ const docsLink = "https://developers.drata.com/docs/openapi/reference/operation/
 export default {
   key: "drata-vendor-updated",
   name: "Vendor Updated",
-  description: `Emit new event when a vendor is updated. [See docs here](${docsLink}).`,
+  description: `Emit a new event when a vendor is updated. [See the documentation](${docsLink}).`,
```
Comment (dylburger) [inline components/drata/sources/vendor-updated/vendor-updated.mjs:10]:
```suggestion
  description: `Emit an event when a vendor is updated. [See the documentation](${docsLink}).`,
```

Comment (dylburger) [reply components/drata/sources/vendor-updated/vendor-updated.mjs:10]: Removed "new" for consistency with Google's Developer docs style guide

**Review by feyzullah — APPROVED** [review]

**Pull request merged**